src/cron.ts

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-console */
+import { CronJob } from 'cron';
+import { client, e } from './lib/db';
+
+const proceedingCompletedStates = [
+    'complaintSent',
+    'secondAnalysisFoundNothing',
+    'secondAnalysisFailed',
+    'initialAnalysisFoundNothing',
+    'initialAnalysisFailed',
+    'expired',
+    'erased',
+];
+
+// Garbage collection
+new CronJob('08 12 03 * * *', async () => {
+    console.log('Running garbage collection...');
+
+    // Proceedings expired after one year.
+    const oneYearAgo = new Date();
+    oneYearAgo.setFullYear(oneYearAgo.getFullYear() - 1);
+
+    const expiredProceedings = await e
+        .update(e.Proceeding, (p) => ({
+            filter: e.all(
+                e.set(
+                    e.op(p.state, 'not in', e.set(...proceedingCompletedStates)),
+                    e.op(p.createdOn, '<=', oneYearAgo),
+                ),
+            ),
+
+            set: { expired: e.datetime_current() },
+        }))
+        .run(client);
```

Comment:
Shouldn’t we also delete the uploads here immediately and not just another week later?